worker/rhpv3.go

```diff
@@ -837,6 +837,191 @@ func RPCReadRegistry(t *rhpv3.Transport, payment rhpv3.PaymentMethod, key rhpv3.
 	}, nil
 }
 
+// Renew renews a contract with a host. To avoid an edge case where the contract
```

Comment:
Unfortunately it's not really a "just pay for it" here. The contract we have might not have funds in it and we may not have a valid price table with the host.

So the flow would be something like 

1. check if we have a pricetable
2. if we don't, check if we have a funded account to fetch one
3. if we don't, check if we have money in the contract to fetch one
4. if we don't, don't pay

To me it seemed like a worthwhile compromise to default to not paying a few Hastings every 3 months or so to avoid all of that.